destination/keepsafe.go

```diff
@@ -43,8 +43,8 @@ func (k *keepSafe) Add(buf []byte) {
 func (k *keepSafe) GetAll() [][]byte {
 	k.Lock()
 	ret := append(k.safeOld, k.safeRecent...)
-	k.safeOld = make([][]byte, 0, k.initialCap)
-	k.safeRecent = make([][]byte, 0, k.initialCap)
+	k.safeOld = k.safeOld[:0]
```

Comment:
this is unsafe because it will store data into the same backing array as the one which is being returned which will be read from by the caller of GetAll().
what's the goal of these lines changing? just a memory-reuse optimization I guess, but doesn't look related to the mem leak itself.